dot/core/messages.go

```diff
@@ -70,32 +74,34 @@ func (s *Service) HandleTransactionMessage(peerID peer.ID, msg *network.Transact
 		return false, err
 	}
 
-	allTxnsAreValid := true
+	allTxsAreValid := true
 	for _, tx := range txs {
-		txnIsValid := true
+		isValidTxn := true
 		validity, err := s.validateTransaction(head, rt, tx)
 		if err != nil {
-			txnIsValid = false
-			allTxnsAreValid = false
-			switch err := err.(type) {
+			switch err.(type) {
 			case runtime.InvalidTransaction:
+				isValidTxn = false
+				allTxsAreValid = false
 				s.net.ReportPeer(peerset.ReputationChange{
 					Value:  peerset.BadTransactionValue,
 					Reason: peerset.BadTransactionReason,
 				}, peerID)
 			case runtime.UnknownTransaction:
+				isValidTxn = false
+				allTxsAreValid = false
 			default:
-				return false, fmt.Errorf("validating transaction from peerID %s: %w", peerID, err)
+				return false, fmt.Errorf("failed validating transaction for peerID %s: %w", peerID, err)
 			}
 		}
 
-		if txnIsValid && validity.Propagate {
+		if isValidTxn && validity.Propagate {
```

Comment:
using the `continue` in the following cases: `runtime.InvalidTransaction`, `runtime.UnknownTransaction` you don't need to have `isValidTxn` because you will jump to the next iteration item without passing through this if statement

```suggestion
		if validity.Propagate {
```

Reply:
Nice!